services/src/main/java/org/keycloak/protocol/oidc/mappers/OIDCAttributeMapperHelper.java

```diff
@@ -309,8 +311,24 @@ private static void mapClaim(List<String> split, Object attributeValue, Map<Stri
         int i = 0;
         for (String component : split) {
             i++;
-            if (i == length) {
+            if (i == length && length == 1 && primaryClaims.contains(split.get(0))) {
```

Comment:
Instead of using the hard-coded `primaryClaims` list maybe we can better rely on the multi-valued configuration option. Those primary claims will never be defined as multi (and if they are they are probably wrong defined). You can pass the model to the method because it's private without problems.

@cgeorgilakis WDYT? Do you see any problem with that approach?